### ISSUE TITLE
TCH004 incorrect edge case

### ISSUE THREAD

**Issue by nstarman** [charliermarsh/ruff#2895]
I found an incorrect detection of ``TCH004``. MWE:

```
from typing import Any, TYPE_CHECKING

if TYPE_CHECKING:
    from numpy import floating
    from numpy.typing import NDArray
    from typing_extensions import TypeAlias

    NDFloating: TypeAlias = NDArray[floating[Any]]

def func(x: NDFloating) -> None:
    return
```

ruff is saying ``floating`` and ``NDArray`` should be moved out of the type-checking block because the import is used for more than type hinting. However ``NDFloating`` is constructed in the TYPE_CHECKING block, so I think this is a false positive.

**Comment by charliermarsh** [comment]
I.e., I _think_ this is the same as #2852.
